Matrix.java

```diff
@@ -0,0 +1,89 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.function.BinaryOperator;
+
+public class Matrix<T> {
+
+    private int rows;
+    private int columns;
+    private LinkedList<LinkedList<T>> matrix;
+
+
+    public Matrix(int rows, int columns, Scanner scanner) throws FileNotFoundException {
```

Comment:
You don't want constructors throwing exceptions if you can avoid it. Nor do you want reading files in constructors --- it's best if a function does only one thing, and the constructor is already constructing the object in memory.

A better design would be to have a static factory method that takes the name of the file / scanner, and returns the constructed object. While the constructor would have some "normal" arguments, say, arraylist of arraylists.

It would also simplify matters when writing unit tests --- you will not need a disk file to write tests that need Matrix creation.